File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/card/DeleteCardUseCase.java
package com.lattis.ellipse.domain.interactor.card;

import com.lattis.ellipse.data.network.model.response.BasicResponse;
import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.UseCase;
import com.lattis.ellipse.domain.repository.CardRepository;

import javax.inject.Inject;

import io.reactivex.Observable;


public class DeleteCardUseCase extends UseCase<BasicResponse> {

    private CardRepository cardRepository;


    public DeleteCardUseCase setCardId(int cardId) {
        this.cardId = cardId;
        return this;
    }

    private int cardId;


    @Inject
    public DeleteCardUseCase(ThreadExecutor threadExecutor,
                             PostExecutionThread postExecutionThread,
                             CardRepository cardRepository) {
        super(threadExecutor, postExecutionThread);
        this.cardRepository = cardRepository;
    }


    @Override
    protected Observable<BasicResponse> buildUseCaseObservable() {
        return cardRepository.deleteCard(cardId);
    }


}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/card/DeleteCardUseCase.kt
package com.lattis.domain.usecase.card

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.repository.CardRepository
import com.lattis.domain.usecase.base.UseCase
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject

class DeleteCardUseCase @Inject constructor(
    threadExecutor: ThreadExecutor,
    postExecutionThread: PostExecutionThread,
    private val cardRepository: CardRepository
) : UseCase<Boolean>(threadExecutor, postExecutionThread) {
    private var cardId = 0
    fun setCardId(cardId: Int): DeleteCardUseCase {
        this.cardId = cardId
        return this
    }
    override fun buildUseCaseObservable(): Observable<Boolean> {
        return cardRepository.deleteCard(cardId)
    }

}